libqf/libqfqmlwidgets/src/spinbox.cpp

```diff
@@ -34,3 +34,32 @@ void SpinBox::onValueChanged(int n)
 	saveDataValue();
 	emit dataValueChanged(n);
 }
+
+void SpinBox::keyPressEvent(QKeyEvent * event) {
+	QString text;
+	auto code = event->nativeScanCode();
+#ifdef __unix__
+	// Xorg has an offset of 8 due to historical reasons
+	code = code - 8;
+#endif
+	switch (code) {
+		case 2:
+		case 3:
+		case 4:
+		case 5:
+		case 6:
+		case 7:
+		case 8:
+		case 9:
+		case 10:
+			text = QString::number(code - 1);
+			break;
+		case 11:
+			text = "0";
+			break;
+		default:
+			text = event->text();
+	}
+	QKeyEvent * e = new QKeyEvent(event->type(), event->key(), event->modifiers(), text, event->isAutoRepeat(), event->count());
```

Comment:
I think that this is a memory leak, you should IMO set value of spinbox here instead of creating fake event

Reply:
not faking the event would require non trivial reimplementation of the Qt functionality. I admit there might be a memory leak though.